### DISCUSSION TITLE
Tap Tempo fails

### DISCUSSION THREAD

**Discussion by Djard** [hydrogen-music/hydrogen#1673] [Hydrogen on Windows]
I faithfully followed the instructions in the manual to use the Tap Tempo feature, without success.  Below are the steps I used. Perhaps someone can correct the steps or add whatever is missing.

1. Select "Song" in play controls section.
2. Select "Song" mode -> click in "Timeline" a few bars before the section you want to modify.
3. Play -> tap on <kbd>\\</kbd> key just before bar # 12, gradually slowing to target speed -> click on "Stop" button.

TIP: To facilitate orientation, tap the <kbd>\\</kbd> key lightly, until just before the point where you want to alter the tempo then fully tap the key.

4. Click in Timeline before the section -> Play (to audit your work). If not satisfied, click on "Undo" and try again.

**Comment by Djard** [comment]
This page adds a link in step 3 above, which I was unable to delete: it should read...

3. Play -> tap on key just before bar No. 12 (the bar in which I want to slow down the tempo).

**Reply by theGreatWhiteShark** [reply]
> This page adds a link in step 3 above, which I was unable to delete

I've updated it and removed the link. `#` + number automatically links issues, discussion, and pull requests within Github.

**Comment by theGreatWhiteShark** [comment]
Hey @Djard ,

I'm not a 100% sure what you are trying to accomplish / how do you expect the tap tempo to work.

Tempo handling in Hydrogen is a bit weird and not quite intuitive. In the [upcoming](https://ci.appveyor.com/api/buildjobs/fxbnupbdd6ec4lt9/artifacts/Hydrogen-1.1.1-win64.exe) version hopefully things will be more clear. There all tempi set using tap tempo, beat counter, the tempo widget in the player control, or MIDI and OSC commands are affecting the overall tempo of the whole song. The Timeline, on the other hand, will live in a different domain and once activated none of the above ways to alter tempo will take effect before deactivating the Timeline again. So, one can either adjust tempo freely and independent of the playback position **or** use tempo markers to alter tempo at predefined locations. This way jumping back and forth within the song is deterministic and there are now synchronization issues with external apps anymore.

Using tap tempo will thus overwrite the song's current tempo and there is no way to reset with other than undoing the changes, setting the original tempo manually, or reopening the song without saving the changes.

**Reply by Djard** [reply]
I was trying to create a gradual slowing of tempo in the last measure of a project (i.e. add a _rallentando_). Do I understand you correctly that Tap Tempo and Beat Counter work to modify the BPM of the whole song and cannot be used to alter the BPM of just a section?

**Reply by theGreatWhiteShark** [reply]
> Do I understand you correctly that Tap Tempo and Beat Counter work to modify the BPM of the whole song and cannot be used to alter the BPM of just a section?

Yes. You can imagine it as two different modes: with Timeline activated you can program tempo changes which will be picked up automatically. Without Timeline (and using e.g. tap tempo) you can freely alter tempo regardless of the song position but you have to do so manually.

I'm afraid the feature you want is a gradual tempo change programmed using the Timeline which is not supported yet.

**Reply by Djard** [reply]
A belated thank you.

**Comment by Djard** [comment]
Thanks. You have saved me much time in experimenting.
Hydrogen is a powerful program and fun to use, but without the ability to add an accel. or rall.-- quite common in all genres of music--it remains limited to basic rhythms.
Until the feature is added, I will continue laboring with Ableton to create drum tracks.
Again, thank you for lending me a hand.



Sent from Yahoo Mail on Android 
 
  On Tue, Nov 15, 2022 at 1:30 PM, ***@***.***> wrote:   



Do I understand you correctly that Tap Tempo and Beat Counter work to modify the BPM of the whole song and cannot be used to alter the BPM of just a section?


Yes. You can imagine it as two different modes: with Timeline activated you can program tempo changes which will be picked up automatically. Without Timeline (and using e.g. tap tempo) you can freely alter tempo regardless of the song position but you have to do so manually.

I'm afraid the feature you want is a gradual tempo change programmed using the Timeline which is not supported yet.
